include/aws/common/json.h

```diff
@@ -211,6 +236,29 @@ size_t aws_json_get_array_size(const struct aws_json_value *array);
  */
 AWS_COMMON_API
 int aws_json_value_remove_array_element(struct aws_json_value *array, size_t index);
+
+/**
+ * @brief callback for iterating elements of an array.
+ * return false to exit early.
+ */
+typedef bool(aws_json_on_element_encountered_const_fn)(size_t key, const struct aws_json_value *value, void *user_data);
+
+/**
+ * @brief iterates though elements of an array.
+ * iteration is sequential starting with 0th element.
+ * likely to be faster than manually calling get element at.
+ * @param array array to iterate over.
+ * @param on_key callback for when element is endcountered.
+ * @param user_data user data to pass back in callback.
+ * @return AWS_OP_SUCCESS when iteration finishes completely or exits early,
+ * AWS_OP_ERR if value is not an array.
+ */
+AWS_COMMON_API
+int aws_json_const_iterate_array(
+    struct aws_json_value *array,
+    aws_json_on_element_encountered_const_fn *on_element,
```

Comment:
"on_value" probably a better name than "on_element" for the typedef and param.

This is is more debatable than my comment on the object iterator.

While json.org does say an "array" is composed of "elements", an "element" is "ws value ws" and we don't super care about whitespace so 🤷‍♀️

Reply:
I actually like on value here. reduces the cognitive load by not introducing another concept